### PR TITLE
Better docs for HCL

### PR DESCRIPTION
**Pull request by azr**
This is a draft PR on HCL docs. Help me document this if you will ! 🙂

### CODE REVIEW

File: website/source/guides/hcl/component-object-spec/index.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+layout: guides
+page_title: Generating code for congig spec.
```
Comment (nywilken) [inline website/source/guides/hcl/component-object-spec/index.html.md.erb]:
```suggestion
page_title: Generating code for config spec.
```

File: website/source/guides/hcl/component-object-spec/index.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+* Make sure that all the nested fields of Config are auto generated the same way.
+
+* Now we only need to make Builder implement the interface by adding the
+following snippet:
+
+    ```go
+    func (b *Builder) ConfigSpec() hcldec.ObjectSpec { return b.config.FlatMapstructure().HCL2Spec() }
+    ```
+
+    From now one every time you add or change a field of Config you will need to
```
Comment (nywilken) [inline website/source/guides/hcl/component-object-spec/index.html.md.erb]:
```suggestion
    From now on every time you add or change a field of Config you will need to
```

File: website/source/guides/hcl/component-object-spec/index.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    run he `go generate` command again.
```
Comment (nywilken) [inline website/source/guides/hcl/component-object-spec/index.html.md.erb]:
```suggestion
    run the `go generate` command again.
```

File: website/source/guides/hcl/from-json-v1/index.html.md.erb [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 page_title: Transforming Packer v1 files for Packer v1.5.0
-sidebar_current: hcl
+sidebar_current: hcl-from-json-v1
 description: |-
   Learn how to manually move from a Packer v1 working JSON build file to a
   working v1.5.0 HCL file.
 ---
 
-# Transforming Packer v1 files for Packer v1.5
+# Transforming Packer v1 config files for Packer v1.5
```
Comment (nywilken) [inline website/source/guides/hcl/from-json-v1/index.html.md.erb]:
```suggestion
# Transforming Packer v1 config files to HCL2 for Packer v1.5
```

**Comment by azr** [comment]
Okay, this looks good to me tell me what you think. A netlify preview is available. 

https://deploy-preview-8501--packer-www.netlify.com/guides/hcl
https://deploy-preview-8501--packer-www.netlify.com/docs/configuration/from-1.5/syntax.html
https://deploy-preview-8501--packer-www.netlify.com/docs/configuration/from-1.5/expressions.html
https://deploy-preview-8501--packer-www.netlify.com/docs/configuration/from-1.5/syntax-json.html

https://deploy-preview-8501--packer-www.netlify.com/docs/configuration/from-1.5/index.html
https://deploy-preview-8501--packer-www.netlify.com/docs/configuration/from-1.5/syntax.html
https://deploy-preview-8501--packer-www.netlify.com/docs/configuration/from-1.5/expressions.html
https://deploy-preview-8501--packer-www.netlify.com/docs/configuration/from-1.5/syntax-json.html

**Pull request merged**